### ISSUE TITLE
Is there a way to set the pointer array of image directly instead of constructing a PackedByteArray to create an image?

### ISSUE THREAD

**Issue by ooyyloo** [godot-rust/gdext#825]
I'm trying to write a program that read images and show them. I use OpenCV to read image files. To show images, I have to convert bytes of image to a PackedByteArray. This costs more than 1 second. It's slow, and can't be endured. When I do not create image, it just costs less than 500ms(read file and cvtColor). 
So I'm asking here, to find a solution of show image files quickly.
The reason I don't use Image.load is that, the file types of image supported by godot can't meet my requirements.

**Comment by Bromeon** [comment]
You should fill the `PackedByteArray` as early as possible in the chain, avoiding expensive copies each frame.
Furthermore, try to reuse the same packed array across frames. It should not trigger CoW if Godot doesn't keep hold of the buffer.

`PackedByteArray` cannot point to foreign memory, it needs to own it due to the CoW semantics.

Ultimately, we're limited by the APIs that Godot provides, so this is more a Godot instead of godot-rust related question.

**Comment by ooyyloo** [comment]
After searching for some code in godot, I think the PackedByteArray does not matter in construction of Image object. A pointer and the size are enough.
From the image below, we can know data of godot image was passed by passing a reference named p_data. The reference in C++ is equal to pointer variable ("reference" can be transfromed to "pointer"). So, it's not a matter of godot APIs, but a matter of gdext.
![image](https://github.com/user-attachments/assets/22937ef4-7723-47b4-9e8d-832b5b42767c)

**Comment by Bromeon** [comment]
And which *public* Godot API allows passing a pointer? We don't have access to Godot's internals.

Also, why upvote your own comment? 🤔

**Comment by ooyyloo** [comment]
> And which _public_ Godot API allows passing a pointer? We don't have access to Godot's internals.
> 
> Also, why upvote your own comment? 🤔

![image](https://github.com/user-attachments/assets/eda92b56-6aac-4270-8aa1-f5e4bf86e84a)
Isn't this function a public Godot API? The p_data parameter, can just be passed as * *vector_ptr *.

**Comment by Bromeon** [comment]
> Isn't this function a public Godot API?

Well no, the public API is documented here:
https://docs.godotengine.org/en/stable/classes

It only has the function you already discovered, which takes `PackedByteArray`:
![image](https://github.com/user-attachments/assets/cc733c90-42e7-4b68-8812-774d538f647f)

----

TLDR: you have a fair point, but you should bring this up with Godot.  We cannot address this on Rust side.

There seems to be a proposal already:
https://github.com/godotengine/godot-proposals/issues/868
It hasn't seen any activity for quite a while, so feel free to write a comment and elaborate your use case.

**Comment by Bromeon** [comment]
Closing as answered. Feel free to reference this issue in any potential Godot discussion. Thanks for bringing it up!
